drools-wb-screens/drools-wb-appformer-extensions-editor/drools-wb-appformer-extensions-editor-backend/src/main/java/org/drools/workbench/screens/appformer/extensions/backend/PerspectiveExtensionServiceImpl.java

```diff
@@ -0,0 +1,223 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.appformer.extensions.backend;
+
+import java.util.List;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.drools.workbench.screens.appformer.extensions.model.PerspectiveExtensionContent;
+import org.drools.workbench.screens.appformer.extensions.service.PerspectiveExtensionService;
+import org.guvnor.common.services.backend.config.SafeSessionInfo;
+import org.guvnor.common.services.backend.exceptions.ExceptionUtilities;
+import org.guvnor.common.services.backend.util.CommentedOptionFactory;
+import org.guvnor.common.services.backend.validation.GenericValidator;
+import org.guvnor.common.services.shared.metadata.model.Metadata;
+import org.guvnor.common.services.shared.metadata.model.Overview;
+import org.guvnor.common.services.shared.validation.model.ValidationMessage;
+import org.jboss.errai.bus.server.annotations.Service;
+import org.kie.workbench.common.services.backend.service.KieService;
+import org.uberfire.backend.server.util.Paths;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.ext.editor.commons.service.CopyService;
+import org.uberfire.ext.editor.commons.service.DeleteService;
+import org.uberfire.ext.editor.commons.service.RenameService;
+import org.uberfire.ext.layout.editor.api.LayoutServices;
+import org.uberfire.ext.layout.editor.api.editor.LayoutTemplate;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.file.FileAlreadyExistsException;
+import org.uberfire.rpc.SessionInfo;
+import org.uberfire.workbench.events.ResourceOpenedEvent;
+
+@Service
+@ApplicationScoped
+public class PerspectiveExtensionServiceImpl
+        extends KieService<PerspectiveExtensionContent>
+        implements PerspectiveExtensionService {
+
+    @Inject
+    @Named("ioStrategy")
+    private IOService ioService;
+
+    @Inject
+    private CopyService copyService;
+
+    @Inject
+    private DeleteService deleteService;
+
+    @Inject
+    private RenameService renameService;
+
+    @Inject
+    private Event<ResourceOpenedEvent> resourceOpenedEvent;
+
+    @Inject
+    private GenericValidator genericValidator;
+
+    @Inject
+    private CommentedOptionFactory commentedOptionFactory;
+
+    private SafeSessionInfo safeSessionInfo;
+
+    @Inject
+    private LayoutServices layoutServices;
+
+    public PerspectiveExtensionServiceImpl() {
+    }
+
+    @Inject
```

Comment:
Best to be consistent with injection; either field or constructor (and constructor is going to make your life easier for Unit Testing.. which, given the field instances are ```private``` I suspect there's no tests for this class.. but I'll find out soon enough).